api/control/v1alpha1/types.go

```diff
@@ -490,8 +490,24 @@ type ReturnedState struct {
 type PropagationData struct {
 	// `lastReturnedUpdateTimestamp` is the time of the last update to any
 	// of the returned object state in the core.
-	// Before the first such update, this is the zero value of `time.Time`.
+	// Before the first such update, this holds the zero value of `time.Time`.
 	LastReturnedUpdateTimestamp metav1.Time `json:"lastReturnedUpdateTimestamp"`
+
+	// `lastGeneration` is that last `ObjectMeta.Generation` from the WDS that
+	// propagated to the WEC. This is not to imply that it was successfully applied there;
+	// for that, see `lastGenerationIsApplied`.
+	// Zero means that none has yet propagated there.
+	LastGeneration int64 `json:"lastGeneration"`
+
+	// `lastGenerationIsApplied` indicates whether `lastGeneration` has been successfully
+	// applied in the WEC. This is `false` while no attempt has yet been made.
+	LastGenerationIsApplied bool `json:"lastGenerationIsApplied"`
```

Comment:
what about the situation where an attempt has been made but was not successful?
does it considered as applied or not? 
this is not clear from the comment.

Reply:
Do you see the qualifier "successfully" in the first sentence of the comment?

Reply:
I did, the last sentence got me confused.

Reply:
Since it confused you and @vMaroon, I have removed it.